### ISSUE TITLE
Find All References: invalid 'no references found' message when other references exist

### ISSUE THREAD

**Issue by zanaptak** [dotnet/fsharp#9984]
(Issue extracted from #9924. Reproduces on commit 14a29caa864707da0ddd60de93e1a4e9eae67b15.)

![image](https://user-images.githubusercontent.com/53196138/90938430-b228ae00-e3ce-11ea-8cfd-f146edf912ed.png)

**Repro steps**

- Clone: https://github.com/zanaptak/references.git
- Open solution
- Open View.fs
- Find All References on `Increment`

**Related information**

* Operating system: Windows 10
* .NET Runtime kind (.NET Core, .NET Framework, Mono): .NET Core SDK 3.1.401
* Editing Tools (e.g. Visual Studio Version, Visual Studio): Visual Studio 16.7.1 + commit 14a29caa864707da0ddd60de93e1a4e9eae67b15

**Comment by jmagaram** [comment]
I am having same issue - search for find references and it doesn't find anything. But if I go to the reference and choose "go to definition" it finds it. In general I'm finding F# on VS 2019 very buggy. Rename refactor works maybe 20% of the time and the other times when I start typing the new refactored name it just changes it there - hmmm... maybe this is related to this "find references" bug. Don't have a repro scenario. Seems flaky. Have experienced this for weeks.

**Comment by cartermp** [comment]
> search for find references and it doesn't find anything

Note that what you're describing is a different problem. In this case it does find all references but there is an erroneous display in the window. What you're describing there is likely resolved in https://github.com/dotnet/fsharp/pull/9934 and will release with VS 16.8.

> Rename refactor works maybe 20% of the time and the other times when I start typing the new refactored name it just changes it there

This is a separate issue I think. Just to double-check, are all your files saved when you invoke Rename? Sadly it is required otherwise it'll mess up symbols getting renamed. There is a deep-rooted assumption in the compiler that files are saved on disk for Rename. We're currently working through that particular issue as it's dreadful, but unfortunately not something a quick fix can be applied to.

**Comment by jmagaram** [comment]
I'm seeing the problem of finding references not finding anything with 16.8.0 preview 2.1. Should it be fixed with that? My cursor is on "current" at the top of the image, but find references doesn't find the item just below it. In this case, if I try to refactor rename it doesn't work. Also see photo. Does seem related bug - if it can't find the reference it probably can't do the rename elsewhere. The file is saved to disk.

![image](https://user-images.githubusercontent.com/14130251/92204593-a90cf780-ee38-11ea-9235-97e37ed1cd6c.png)

![image](https://user-images.githubusercontent.com/14130251/92204701-e8d3df00-ee38-11ea-8fc1-63e2683810c0.png)

**Comment by cartermp** [comment]
@jmagaram No, but you may say the fix in either 16.8 preview 3 or preview 4. In this case it might seem that the rename issue you're seeing is related then.

**Comment by cartermp** [comment]
Note another trivial example can be found here: https://github.com/dotnet/fsharp/issues/10690

Define the following F# class:

```fsharp
type C() =
    member val InViewport = false with get, set 
```

Use the member val in another file:

```fsharp
let c = C()
c.InViewport <- true
```

If you do Find Refs from the definition, it doesn't find anything. If you do it from a usage, it finds the usage.

Also note that rename does not pick up the symbol use across files.

cc @tihan
